lib/livebook_web/endpoint.ex

```diff
@@ -11,11 +11,16 @@ defmodule LivebookWeb.Endpoint do
     same_site: "Lax"
   ]
 
-  socket "/live", LivebookWeb.Socket,
-    # Don't check the origin as we don't know how the web app is gonna be accessed.
-    # It runs locally, but may be exposed via IP or domain name.
-    # The WebSocket connection is already protected from CSWSH by using CSRF token.
-    websocket: [check_origin: false, connect_info: [:user_agent, :uri, session: @session_options]]
+  # Don't check the origin as we don't know how the web app is gonna be accessed.
+  # It runs locally, but may be exposed via IP or domain name. The WebSocket
+  # connection is already protected from CSWSH by using CSRF token.
+  @websocket_options [
+    check_origin: false,
+    connect_info: [:user_agent, :uri, session: @session_options]
```

Comment:
For completeness, there is another approach for authenticating the `LivebookWeb.Socket` which is to pass a Phoenix.Token with the relevant information and we then decode on channel join.

Reply:
Yeah, I did it for consistency with the other socket and to reuse the existing auth logic :)